XiEditor/CoreConnection.swift

```diff
@@ -39,24 +39,37 @@ struct FileWriter {
             return nil
         }
     }
-
+    
     func write(bytes: Data) {
         handle.write(bytes)
     }
 }
 
 class CoreConnection {
-
+    
+    let task = Process()
+    var timer: Timer?
     var inHandle: FileHandle  // stdin of core process
     var recvBuf: Data
     weak var client: XiClient?
     let rpcLogWriter: FileWriter?
+    let errLogWriter: FileWriter?
+    
+    // default log directory on MacOS is /Library/Logs
+    let logDirectory = FileManager.default.urls(
```

Comment:
we've already declared this in the app delegate; we should only declare it once. (App delegate is probably the reasonable place, and we can access the app delegate vae with `(NSApplication.shared.delegate as! AppDelegate).errorLogDirectory`

Reply:
Agreed.  Let's unify the path computation.